crates/s3s/src/validation.rs

```diff
@@ -0,0 +1,64 @@
+//! Validation API for S3 bucket names
+
+/// Trait for validating S3 names
+///
+/// Implementations should return `true` for valid names and `false` for invalid ones.
+pub trait NameValidation: Send + Sync {
+    /// Validate a bucket name
+    fn validate_bucket_name(&self, name: &str) -> bool;
+}
+
+/// Default AWS-compliant name validation
+#[derive(Debug, Clone, Default)]
+pub struct DefaultNameValidation;
+
+/// Static instance of `DefaultNameValidation` to avoid allocations
+pub static DEFAULT_NAME_VALIDATION: DefaultNameValidation = DefaultNameValidation;
```

Comment:
We can create an instance during `call` function. There is no allocation since `DefaultNameValiation` is a struct type on stack (zero-size for now). 
So this line is unnecessary.